### PR TITLE
Compare config vars to trigger an app restart

### LINKED ISSUE #286: Separating config variables breaks app restart

### PR DESCRIPTION
**Pull request by pcarranzav**
Fixes #286

### CODE REVIEW

File: src/application.coffee [hunk 1]
```diff
@@ -551,7 +551,9 @@ compareForUpdate = (localApps, remoteApps) ->
 	toBeUpdated = _.filter toBeUpdated, (appId) ->
 		localApp = _.omit(localApps[appId], 'config')
 		remoteApp = _.omit(remoteApps[appId], 'config')
-		return !_.isEqual(remoteApp, localApp)
+		return !_.isEqual(remoteApp, localApp) or
+			localApps[appId].config?.RESIN_DEVICE_RESTART != remoteApps[appId].config?.RESIN_DEVICE_RESTART or
```
Comment (Page-) [inline src/application.coffee]: Maybe something like
```coffee
restartVars = (config) ->
	return _.pick(config, 'RESIN_DEVICE_RESTART', 'RESIN_RESTART')
!_.isEqual(restartVars(localApps[appId].config), restartVars(remoteApps[appId].config))
```
?

Comment (pcarranzav) [inline src/application.coffee]: Oooh, nice one

**Comment by pcarranzav** [comment]
@Page- made an extra change to prevent update strategies from turning a simple restart into something more complex. Still approve?

File: src/application.coffee [hunk 2]
```diff
@@ -613,14 +621,16 @@ application.update = update = (force) ->
...
+						else if _.includes(toBeUpdated, appId) or _.includes(toBeRestarted, appId)
 							app = remoteApps[appId]
 							conf = JSON.parse(app.config)
 							forceThisApp =
 								conf['RESIN_SUPERVISOR_OVERRIDE_LOCK'] == '1' ||
 								conf['RESIN_OVERRIDE_LOCK'] == '1'
 							strategy = conf['RESIN_SUPERVISOR_UPDATE_STRATEGY']
 							timeout = conf['RESIN_SUPERVISOR_HANDOVER_TIMEOUT']
+							if _.includes(toBeRestarted, appId) and strategy is 'delete-then-download'
+								strategy = 'download-then-kill'
```
Comment (Page-) [inline src/application.coffee]: Can you add a comment on the reasoning for this please

Comment (pcarranzav) [inline src/application.coffee]: Just realized this should actually be fixed in the strategy itself, as it also affects updates that don't require a download.

**Pull request merged**